concent_api/core/tests/test_unit_authentication_utilities.py

```diff
@@ -0,0 +1,214 @@
+from django.test                    import override_settings
+
+from golem_messages                 import message
+from golem_messages.shortcuts       import dump
+from golem_messages.shortcuts       import load
+
+from core.exceptions                import Http400
+from core.validation                import validate_golem_message_client_authorization
+from core.validation                import validate_golem_message_signed_with_key
+from core.validation                import validate_list_of_identical_task_to_compute
+from core.tests.utils               import ConcentIntegrationTestCase
+from utils.shortcuts                import load_without_public_key
+from utils.testing_helpers          import generate_ecc_key_pair
+
+
+(CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY) = generate_ecc_key_pair()
+
+
+@override_settings(
+    CONCENT_PRIVATE_KEY       = CONCENT_PRIVATE_KEY,
+    CONCENT_PUBLIC_KEY        = CONCENT_PUBLIC_KEY,
+)
+class LoadWithoutPublicKeyUnitTest(ConcentIntegrationTestCase):
+
+    def test_load_without_public_key_should_load_message(self):
+        """
+        Test that message loaded with load_without_public_key function will be the same as load
+        with golem_messages load function.
+        """
+
+        # Create and fill some data into ComputeTaskDef
+        compute_task_def = message.ComputeTaskDef()
+        compute_task_def['task_id']     = '1'
+        compute_task_def['subtask_id']  = '2'
+        compute_task_def['deadline']    = 1510912800
+
+        # Create TaskToCompute
+        task_to_compute = message.TaskToCompute(compute_task_def = compute_task_def)
+
+        # Dump TaskToCompute to make it signed
+        dumped_task_to_compute = dump(task_to_compute, self.REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+
+        loaded_task_to_compute_with_utility_function    = load_without_public_key(
+            dumped_task_to_compute,
+        )
+
+        loaded_task_to_compute_with_golem_messages_load = load(
+            dumped_task_to_compute,
+            CONCENT_PRIVATE_KEY,
+            self.REQUESTOR_PUBLIC_KEY,
+        )
+
+        self.assertEqual(loaded_task_to_compute_with_utility_function, loaded_task_to_compute_with_golem_messages_load)
+
+        for slot in message.TaskToCompute.__slots__:
```

Comment:
is it necessary? is any of the slots differs, would the assertion above pass?

Reply:
I think that extra checking of this won't hurt.

Reply:
I disagree. It distracts you and gets into thinking that somehow these objects cannot be compared normally. It also adds more logic to the test's code which should be avoided (the simpler the tests, the better). If it is not needed, please remove it.

Reply:
done :+1: Ok I have checked how golem_messages are compared and there is a check like this there already.